src/student_success_tool/analysis/pdp/targets/failure_to_graduate_within_x_years.py

```diff
@@ -0,0 +1,213 @@
+    term_is_pre_cohort_col: str = "term_is_pre_cohort",
+    years_to_degree_col: str = "first_year_to_bachelors_at_cohort_inst",
+    enrollment_intensity_col: str = "enrollment_intensity_first_term",
+    term_rank_col: str = "term_rank",
+) -> pd.DataFrame:
+    """
+    Make a labeled dataset for modeling, where each row consists of features
+    from eligible students' first term (within their cohort)
+    matched to computed target variables.
+
+    See Also:
+        - :func:`select_eligible_students()`
+        - :func:`compute_target_variable()`
+        - :func:`shared.get_nth_student_terms()`
+    """
+    df_eligible_students = select_eligible_students(
+        df,
+        student_criteria=student_criteria,
+        student_id_cols=student_id_cols,
+        intensity_time_lefts=intensity_time_lefts,
+        max_term_rank=max_term_rank,
+        num_terms_in_year=num_terms_in_year,
+        n=n,
+        exclude_pre_cohort_terms=exclude_pre_cohort_terms,
+        term_is_pre_cohort_col=term_is_pre_cohort_col,
+        enrollment_intensity_col=enrollment_intensity_col,
+        term_rank_col=term_rank_col,
+    )
+    df_eligible_student_terms = pd.merge(
+        df, df_eligible_students, on=student_id_cols, how="inner"
+    )
+    df_features = shared.get_nth_student_terms(
+        df_eligible_student_terms,
+        student_id_cols=student_id_cols,
+        n=n,
+        sort_cols=term_rank_col,
+        include_cols=None,
+        term_is_pre_cohort_col=term_is_pre_cohort_col,
+        exclude_pre_cohort_terms=exclude_pre_cohort_terms,
+    )
+    df_targets = compute_target_variable(
+        df_eligible_student_terms,
+        intensity_time_lefts=intensity_time_lefts,
+        student_id_cols=student_id_cols,
+        num_terms_in_year=num_terms_in_year,
+        enrollment_intensity_col=enrollment_intensity_col,
+        years_to_degree_col=years_to_degree_col,
+    )
+    df_labeled = pd.merge(df_features, df_targets, on=student_id_cols, how="inner")
+    return df_labeled
+
+
+def compute_target_variable(
+    df: pd.DataFrame,
+    *,
+    intensity_time_lefts: list[tuple[str, float, t.Literal["year", "term"]]],
+    student_id_cols: str | list[str] = "student_guid",
+    num_terms_in_year: int = 4,
+    enrollment_intensity_col: str = "enrollment_intensity_first_term",
+    years_to_degree_col: str = "first_year_to_bachelors_at_cohort_inst",
+) -> pd.Series:
+    """
+    Args:
+        df
+        intensity_time_limits: Mapping of enrollment intensity value (e.g. "FULL-TIME")
+            to the maximum number of years allowed to earn degree "in time" (e.g. 4).
+        student_id_cols
+        enrollment_intensity_col
+        years_to_degree_col: Name of column giving the year in which students _first_
+            earned a particular degree (by default, a Bachelor's degree).
+        num_terms_in_year: Defined number of terms in one academic year.
+    """
+    student_id_cols = utils.to_list(student_id_cols)
```

Comment:
It looks like you landed on something close to what I suggested -- just checking, was that intentional? Last I heard, you found it buggy! :P

Reply:
I had to fix a few things in it, and then I think once I fixed some of the other functions I got it to work!